### ISSUE TITLE
`cargo` runtime crash

### ISSUE THREAD

**Issue by mxcl** [pkgxdev/pantry#3520]
I am on macOS 14, so potentially its ABI breakage in the C++ libs that we didn’t guard against when building cargo.

Once again probs would just be fixed by using rust from rustup

```
cargo --version
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFString stringByStandardizingPath]: unrecognized selector sent to instance 0x600002434050'
*** First throw call stack:
(
	0   CoreFoundation                      0x0000000185fb08c0 __exceptionPreprocess + 176
	1   libobjc.A.dylib                     0x0000000185aa9eb4 objc_exception_throw + 60
	2   CoreFoundation                      0x00000001860623dc -[NSObject(NSObject) __retain_OA] + 0
	3   CoreFoundation                      0x0000000185f1a9e4 ___forwarding___ + 1572
	4   CoreFoundation                      0x0000000185f1a300 _CF_forwarding_prep_0 + 96
	5   Foundation                          0x0000000186fef4f0 -[NSProcessInfo arguments] + 188
	6   CoreFoundation                      0x000000018602bff8 __getDefaultArguments_block_invoke + 96
	7   libdispatch.dylib                   0x0000000185cb1910 _dispatch_client_callout + 20
	8   libdispatch.dylib                   0x0000000185cb314c _dispatch_once_callout + 32
	9   CoreFoundation                      0x000000018602b994 _addBackstopValuesForIdentifierAndSource + 640
	10  CoreFoundation                      0x0000000185ee6254 __81-[_CFXPreferences(SourceAdditions) withNamedVolatileSourceForIdentifier:perform:]_block_invoke + 144
	11  CoreFoundation                      0x000000018602b63c -[_CFXPreferences withNamedVolatileSourceForIdentifier:perform:] + 272
	12  CoreFoundation                      0x0000000185eec5c4 -[CFPrefsSearchListSource addNamedVolatileSourceForIdentifier:] + 136
	13  CoreFoundation                      0x000000018606a8bc __108-[_CFXPreferences(SearchListAdditions) withSearchListForIdentifier:container:cloudConfigurationURL:perform:]_block_invoke.155 + 296
	14  CoreFoundation                      0x000000018606a564 -[_CFXPreferences withSearchLists:] + 84
	15  CoreFoundation                      0x0000000185ee7b18 __108-[_CFXPreferences(SearchListAdditions) withSearchListForIdentifier:container:cloudConfigurationURL:perform:]_block_invoke + 300
	16  CoreFoundation                      0x000000018606a710 -[_CFXPreferences withSearchListForIdentifier:container:cloudConfigurationURL:perform:] + 384
	17  CoreFoundation                      0x0000000185ee7440 -[_CFXPreferences copyAppValueForKey:identifier:container:configurationURL:] + 156
	18  CoreFoundation                      0x0000000185ee7368 _CFPreferencesCopyAppValueWithContainerAndConfiguration + 112
	19  SystemConfiguration                 0x0000000186c77478 SCDynamicStoreCopyProxiesWithOptions + 180
	20  cargo                               0x000000010372be78 Curl_macos_init + 16
	21  cargo                               0x000000010371a69c curl_global_init + 176
	22  cargo                               0x0000000103706b3c _ZN3std4sync4once4Once9call_once28_$u7b$$u7b$closure$u7d$$u7d$17h259c27687f1f5f56E + 36
	23  cargo                               0x000000010387b90c _ZN3std10sys_common4once5queue4Once4call17hdcd5743032d60081E + 756
	24  cargo                               0x0000000103706f04 _ZN4curl9INIT_CTOR9init_ctor17hbf2b5579b3a4a893E + 96
	25  dyld                                0x0000000185b015c8 ___ZZNK5dyld46Loader25findAndRunAllInitializersERNS_12RuntimeStateEENK3$_0clEv_block_invoke + 168
	26  dyld                                0x0000000185b466ec ___ZNK5dyld313MachOAnalyzer18forEachInitializerER11DiagnosticsRKNS0_15VMAddrConverterEU13block_pointerFvjEPKv_block_invoke.202 + 172
	27  dyld                                0x0000000185b39c60 ___ZNK5dyld39MachOFile14forEachSectionEU13block_pointerFvRKNS0_11SectionInfoEbRbE_block_invoke + 496
	28  dyld                                0x0000000185ae12fc _ZNK5dyld39MachOFile18forEachLoadCommandER11DiagnosticsU13block_pointerFvPK12load_commandRbE + 300
	29  dyld                                0x0000000185b38c98 _ZNK5dyld39MachOFile14forEachSectionEU13block_pointerFvRKNS0_11SectionInfoEbRbE + 192
	30  dyld                                0x0000000185b3b780 _ZNK5dyld39MachOFile32forEachInitializerPointerSectionER11DiagnosticsU13block_pointerFvjjRbE + 160
	31  dyld                                0x0000000185b463e0 _ZNK5dyld313MachOAnalyzer18forEachInitializerER11DiagnosticsRKNS0_15VMAddrConverterEU13block_pointerFvjEPKv + 432
	32  dyld                                0x0000000185afd798 _ZNK5dyld46Loader25findAndRunAllInitializersERNS_12RuntimeStateE + 448
	33  dyld                                0x0000000185b03b14 _ZNK5dyld416JustInTimeLoader15runInitializersERNS_12RuntimeStateE + 36
	34  dyld                                0x0000000185afdb4c _ZNK5dyld46Loader23runInitializersBottomUpERNS_12RuntimeStateERN5dyld35ArrayIPKS0_EE + 220
	35  dyld                                0x0000000185b01654 _ZZNK5dyld46Loader38runInitializersBottomUpPlusUpwardLinksERNS_12RuntimeStateEENK3$_1clEv + 112
	36  dyld                                0x0000000185afdccc _ZNK5dyld46Loader38runInitializersBottomUpPlusUpwardLinksERNS_12RuntimeStateE + 304
	37  dyld                                0x0000000185b22ad4 _ZN5dyld44APIs25runAllInitializersForMainEv + 464
	38  dyld                                0x0000000185ae5f34 _ZN5dyld4L7prepareERNS_4APIsEPKN5dyld313MachOAnalyzerE + 3192
	39  dyld                                0x0000000185ae4f44 start + 1948
)
libc++abi: terminating due to uncaught exception of type NSException
zsh: abort      cargo --version
```

**Comment by mxcl** [comment]
Refs https://github.com/rust-lang/cargo/issues/12670

**Comment by jhheider** [comment]
needs #3530

**Comment by mxcl** [comment]
seems to still happen

**Comment by jhheider** [comment]
perhaps we need that -l option to curl.

**Comment by mxcl** [comment]
Fixed by #3546
